syft/workers/socketio_client.py

```diff
@@ -0,0 +1,83 @@
+import time
+from typing import List
+from typing import Union
+
+import socketio
+import torch
+
+import syft as sy
+from syft.frameworks.torch.tensors.interpreters import AbstractTensor
+from syft.workers.virtual import VirtualWorker
+
+
+class WebsocketIOClientWorker(VirtualWorker):
+    """A client which will send a message to a WebsocketIOServerWorker when it receives
+    a message from another worker via _recv_msg
+    This client then waits until the server returns with a result or an ACK at which point
+    it finishes the _recv_msg operation
+    """
+
+    def __init__(
+        self,
+        hook,
+        host: str,
+        port: int,
+        id: Union[int, str] = 0,
+        is_client_worker: bool = False,
+        log_msgs: bool = False,
+        verbose: bool = False,
+        data: List[Union[torch.Tensor, AbstractTensor]] = None,
+    ):
+        self.port = port
+        self.host = host
+        self.uri = f"http://{self.host}:{self.port}"
+
+        self.response_from_client = None
+        self.wait_for_client_event = False
+
+        # Creates the connection with the server
+        self.sio = socketio.Client()
+        super().__init__(hook, id, data, is_client_worker, log_msgs, verbose)
+
+        @self.sio.on("client_ack")
+        def on_client_ack(args):
+            # The server broadcasted an ACK from another client
+            self.response_from_client = b""
+            # Tell the wait_for_client_event to clear up and continue execution
+            self.wait_for_client_event = False
+
+        @self.sio.on("client_send_result")
+        def on_client_result(args):
+            if log_msgs:
+                print("Receiving result from client {}".format(args))
+            # The server broadcasted the results from another client
+            self.response_from_client = args
+            # Tell the wait_for_client_event to clear up and continue execution
+            self.wait_for_client_event = False
+
+    def _send_msg(self, message: bin) -> bin:
+        raise RuntimeError(
+            "_send_msg should never get called on a ",
+            "WebsocketIOClientWorker. Did you accidentally "
+            "make hook.local_worker a WebsocketIOClientWorker?",
+        )
+
+    def _recv_msg(self, message: bin) -> bin:
+        # Sends the message to the server
+        self.sio.emit("message", message)
+        self.wait_for_client_event = True
+        # Wait until the server gets back with a result or an ACK
+        while self.wait_for_client_event:
+            time.sleep(0.1)
+
+        # Return the result
+        if self.response_from_client == b"":
+            return sy.serde.serialize(b"")
```

Comment:
What exactly is this test doing?

Reply:
Also in what cases would this response be returned

Reply:
This represents an empty response. It is happening when the other side (i.e. Android) sends an `ACK`
The deserialisation will not accept something like `"ACK"` thus why it returns an empty response

Reply:
I will make the ACK operation to be `"ACK"` instead of empty up to the point where it's returned to be deserialised

Reply:
Sounds good. That or adding a comment explaining why this is here would work (you should probably add the comment regardless)